README.md

```diff
@@ -4,18 +4,77 @@ A collection of helpers for building a rich text editor (WYSIWYG) with [ProseMir
+### Triggering a new release
 
+This repo uses [changesets](https://github.com/changesets/changesets) for versioning. When you want to trigger a new release:
+- run `yarn changeset add` from your local terminal to generate a changeset.
+    - Select the [semver change type](https://semver.org/) (ie: `major`, `minor`, `patch`)
+    - Write a concise summary of the change -  this is used in the changelog
+- Commit the generated file and merge the PR
+    - This will automatically trigger a release PR
+- Checkout to the release PR (`changeset-release/main`) and push up an empty commit
 ```bash
 git checkout changeset-release/main
 git commit -m "Empty commit to trigger CI tasks" --allow-empty
 git push origin
 ```
+- Approve the PR and merge into `main`
+
+### Why the empty commit?
+
+This is due to a quirk with the PR that is automatically generated. At the moment it cannot trigger the CI tasks to run. In order to run these, push an empty commit to the PR before merging.
+
+
+## Local development
+
+As there is currently no storybook attached to the Rich Text Toolkit, for ease of local development we would recommend either of the following: 
+- run `yarn build` and copy your `/dist` folder into your local `@kaizen/rich-text-editor` package - this is only really helpful to spot check your current change
+- Use [yarn link](https://classic.yarnpkg.com/en/docs/cli/link) or [yacl](https://github.com/wclr/yalc) to create a dependency on your local development environment with the `@kaizen/rich-text-editor`
+  - **Caveat:** there has been some issue getting `yarn link` to work from machine to machine so we recommend using `yacl`
+
+## Using the rich-text-toolkit
+
+The Toolkit was built out with the intention to be used in `@kaizen/rich-text-editor`, which handles the UI and logic around interactive states. While it is recommended that you leverage the Kaizen component, you can utilize the Toolkit to build out a custom solution if modelled similarly.
+
+If you have spoken to the Design System Team and need a bespoke solution, we advise contributing to the Toolkit to build the core functionality and importing that into your project.
+
+
+## Architecture
+
+### Diagram
+
+![RTE Diagram](/docs/assets/rich-text-toolkit-at-a-glance.png)
+*Note: this diagram was created in the planning stage and while it mostly represent the overall structure, there may be some inconsistencies as development continues.*
+
+### Core
+
+Contains the essential functionality to build out the RTE, Including:
+- Initializing the editor view (`/create.ts`)
+    - `createRichTextEditor` creates a new instance of the RTE that can dispatch commands and transactions.
+- Initializing the editor state (`/state.ts`)
+    - `createEditorState` is an API friendly wrapper for ProseMirror's EditorState.create
+    -   `createDocNodeFromContent` creates a Prosemirror doc node by combinining the schema and documents content array
```

Comment:
So I'm reminding myself of these and I actually think none of the stuff in `state.ts` is actually used anywhere. I think I looked at it and decided that there was no need for it to be in separate functions. But I must have forgotten to delete it.

Reply:
oh dang. I will say I have used it in the toolkit tests PR to create a simple instance toolkit, could potentially move it to the helpers folder I made for testing the commands, otherwise will have to update the test to use the hook version

Reply:
oh actually they're used in the tests for createRichTextEditor too, so feels like we should leave them in / potentially move them into the fixtures as a test helper function? I'll axe the callout in the README though

Reply:
Up to you. I was thinking to get rid of them because all they do is call another function. So it just adds a pointless layer IMO.